src/Compilers/Core/Portable/Collections/ImmutableArrayDictionary`2+KeyCollection.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Collections
+{
+    partial struct ImmutableArrayDictionary<TKey, TValue>
+    {
+        public readonly partial struct KeyCollection : IReadOnlyCollection<TKey>, ICollection<TKey>, ICollection
```

Comment:
I'd be tempted to not implement `ICollection<TKey>` and `ICollection`? Then we only have to provide the `GetEnumerator()` to implement `IReadOnlyCollection<TKey>`.

Reply:
➡️ This pull request intentionally exposes the full API surface area.